core/src/main/java/org/apache/brooklyn/core/config/BasicConfigInheritance.java

```diff
@@ -82,9 +223,17 @@ public InheritanceMode isInherited(ConfigKey<?> key, Object from, Object to) {
     
     protected <TContainer, TValue> void checkInheritanceContext(ConfigValueAtContainer<TContainer, TValue> local, ConfigInheritanceContext context) {
         ConfigInheritance rightInheritance = ConfigInheritances.findInheritance(local, context, this);
-        if (!equals(rightInheritance)) 
+        if (!isSameRootInstanceAs(rightInheritance)) {
             throw new IllegalStateException("Low level inheritance computation error: caller should invoke on "+rightInheritance+" "
                 + "(the inheritance at "+local+"), not "+this);
+        }
+    }
+
+    private boolean isSameRootInstanceAs(ConfigInheritance other) {
+        if (other==null) return false;
+        if (this==other) return true;
+        if (other instanceof DelegatingConfigInheritance) return isSameRootInstanceAs( ((DelegatingConfigInheritance)other).getDelegate() );
```

Comment:
Another reason to avoid the delegate. Why is `equals` a bad idea?
Note that you could still end up with an anonymous `BasicConfigInheritance` coming from persistence - they are not resolved to the singletons.